libmachine/host.go

```diff
@@ -276,24 +283,28 @@ func (h *Host) LoadConfig() error {
 		return err
 	}
 
-	// First pass: find the driver name and load the driver
-	var hostMetadata HostMetadata
-	if err := json.Unmarshal(data, &hostMetadata); err != nil {
-		return err
-	}
+	// Remember the machine name and store path so we don't have to pass it
+	// through each struct in the migration.
+	name := h.Name
+	storePath := h.StorePath
 
-	authOptions := hostMetadata.HostOptions.AuthOptions
+	// If we end up performing a migration, we should save afterwards so we don't have to do it again on subsequent invocations.
+	migrationPerformed := h.ConfigVersion != version.ConfigVersion
 
-	driver, err := drivers.NewDriver(hostMetadata.DriverName, h.Name, h.StorePath, authOptions.CaCertPath, authOptions.PrivateKeyPath)
+	migratedHost, err := MigrateHost(h, data)
 	if err != nil {
-		return err
+		return fmt.Errorf("Error getting migrated host: %s", err)
 	}
 
-	h.Driver = driver
+	*h = *migratedHost
```

Comment:
This seems a little ugly / dangerous to me, but I'm not sure if there's a better way unless `LoadConfig` was an external function and not a member of `Host`.